### PR TITLE
Updates documentation

### PR DESCRIPTION
**Pull request by localden**
Follow-up on:

- [X] #2553
- [X] #3847

### CODE REVIEW

File: src/client/Microsoft.Identity.Client/PublicClientApplication.cs [hunk 1]
```diff
@@ -18,11 +18,14 @@ namespace Microsoft.Identity.Client
...
-    /// and as a result have no client secret</description></item>
-    /// <item><description>The redirect URL is proposed by the library. It does not need to be passed in the constructor</description></item>
-    /// </list>
+    /// <para>
+    /// Unlike <see cref="Microsoft.Identity.Client.ConfidentialClientApplication"/>, public clients are unable to hold configuration time secrets,
+    /// and as a result have no client secret.
+    /// </para>
+    /// <para>
+    /// The redirect URI needed for interactive authentication is automatically determined by the library. It does not need to be passed explicitly in the constructor. Depending
+    /// on the authentication strategy (e.g., through the Web Authentication Manager, Authentication app, etc.), different redirect URIs will be used by MSAL.
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/PublicClientApplication.cs:27]:
```suggestion
    /// The redirect URI needed for interactive authentication is automatically determined by the library. It does not need to be passed explicitly in the constructor. Depending
    /// on the authentication strategy (e.g., through the Web Authentication Manager, Authentication app, browser, etc.), different redirect URIs will be used by MSAL. Redirect URIs must always be configured in the AAD portal.
```

**Review by pmaytak — APPROVED** [review]

**Review by neha-bhargava — APPROVED** [review]

File: src/client/Microsoft.Identity.Client/PublicClientApplication.cs [hunk 2]
```diff
@@ -18,11 +18,14 @@ namespace Microsoft.Identity.Client
...
-    /// and as a result have no client secret</description></item>
-    /// <item><description>The redirect URL is proposed by the library. It does not need to be passed in the constructor</description></item>
-    /// </list>
+    /// <para>
+    /// Unlike <see cref="Microsoft.Identity.Client.ConfidentialClientApplication"/>, public clients are unable to hold configuration time secrets,
+    /// and as a result have no client secret.
+    /// </para>
+    /// <para>
+    /// The redirect URI needed for interactive authentication is automatically determined by the library. It does not need to be passed explicitly in the constructor. Depending
+    /// on the authentication strategy (e.g., through the Web Authentication Manager, Authentication app, browser, etc.), different redirect URIs will be used by MSAL. Redirect URIs must always be configured in the Azure Active Directory bldate in the Azure Portal.
```
Comment (gladjohn) [inline src/client/Microsoft.Identity.Client/PublicClientApplication.cs:27]:
```suggestion
    /// on the authentication strategy (e.g., through the Web Authentication Manager, Authentication app, browser, etc.), different redirect URIs will be used by MSAL. Redirect URIs must always be configured in the Azure Active Directory blade in the Azure Portal.
```

**Review by gladjohn — APPROVED** [review]
typo

File: src/client/Microsoft.Identity.Client/PublicClientApplication.cs (hunk 2 above)
Comment (gladjohn) [reply src/client/Microsoft.Identity.Client/PublicClientApplication.cs:27]: also it is in the app registration blade?

**Review by bgavrilMS — APPROVED** [review]

**Comment by gladjohn** [comment]
this PR build isn't running. Does @localden need any other permissions? would be nice to get this merged for the next release.

**Comment by pmaytak** [comment]
> this PR build isn't running. Does @localden need any other permissions? would be nice to get this merged for the next release.

This PR is on a fork and we don't run builds off forks.

**Pull request merged**